### ISSUE TITLE
Thread bursting when the TSDB client object creation failed

### ISSUE THREAD

**Issue by xiaowing** [aliyun/aliyun-tsdb-java-sdk#100]
**Version information**
* SDK version
    v0.3.3

* TSDB engine version
    (none)

**Bug description**
Thread bursting when the TSDB client object creation failed. As a result, it would cause OOM

output as follows

````sh
at first, the active thread number: 2
the current active thread number: 152
the current active thread number: 302
the current active thread number: 452
the current active thread number: 602
the current active thread number: 752
the current active thread number: 902
the current active thread number: 1052
the current active thread number: 1202
the current active thread number: 1352
the current active thread number: 1502
the current active thread number: 1652
the current active thread number: 1802
the current active thread number: 1952
the current active thread number: 2102
the current active thread number: 2252
the current active thread number: 2402
the current active thread number: 2552
the current active thread number: 2702
the current active thread number: 2852
the current active thread number: 3002
the current active thread number: 3152
the current active thread number: 3302
the current active thread number: 3452
the current active thread number: 3602
the current active thread number: 3752
the current active thread number: 3902
the current active thread number: 4052
Exception in thread "pool-1357-thread-1" java.lang.OutOfMemoryError: unable to create new native thread
````

**Reproduce case**
Codes to reproduce the behavior:

````java
public static void testTSDBClientFailure() {
        System.out.printf("at first, the active thread number: %d\n", Thread.activeCount());

        int i = 0;
        while (true) {
            i++;
            try {
                TSDB client = TSDBClientFactory.connect("127.0.0.1", 50010);     // it's an unreachable address
                if (client != null) {
                    System.out.printf("TSDB Client initialized");
                }
            } catch (Exception hex) {
                ;;
            }

            if (i % 50 == 0) {
                System.out.printf("the current active thread number: %d\n", Thread.activeCount());
            }
            try {
                Thread.sleep(50);
            } catch (Exception ex) {
                ;;
            }
        }

    }
````

**Comment by xiaowing** [comment]
经确认，`TSDBClient`的构造函数在确认连接时，如果连接确认失败。虽然关闭了线程池，但是对已启用的Consumer对象（内含3个线程池）没有关闭

**Comment by xiaowing** [comment]
issue fixed by #101
